internal/transform/transform.go

```diff
@@ -140,16 +140,31 @@ func NormalizeSetDirectives(doc *astro.Node) {
 // 	}
 // }
 
-func ExtractScript(doc *astro.Node, n *astro.Node) {
+func ExtractScript(doc *astro.Node, n *astro.Node, opts *TransformOptions) {
 	if n.Type == astro.ElementNode && n.DataAtom == a.Script {
 		if HasSetDirective(n) {
 			return
 		}
 		// if <script hoist>, hoist to the document root
 		// If also using define:vars, that overrides the hoist tag.
 		if hasTruthyAttr(n, "hoist") && !HasAttr(n, "define:vars") {
+			shouldAdd := true
+			for _, attr := range n.Attr {
+				if attr.Key == "src" {
+					if attr.Type == astro.ExpressionAttribute {
+						if opts.StaticExtraction {
+							shouldAdd = false
+							fmt.Printf("%s: <script hoist> uses the expression {%s} on the src attribute and will be ignore. Use a static string on the src attribute instead.\n", opts.Filename, attr.Val)
```

Comment:
nit: `ignored`. 
Also nit: slight preference for `static string literal` or just `string literal`  as more correct here